pkg/sqs/SqsConnectionFactory.php

```diff
@@ -39,11 +39,18 @@ public function __construct($config = 'sqs:')
     {
         if (empty($config) || 'sqs:' === $config) {
             $config = [];
-        } elseif (is_string($config)) {
+        }
+
+        if (is_string($config)) {
             $config = $this->parseDsn($config);
-        } elseif (is_array($config)) {
-        } else {
-            throw new \LogicException('The config must be either an array of options, a DSN string or null');
+        }
+
+        if (!is_array($config)) {
+            throw new \LogicException('The config must be either an array of options, a DSN string, or null');
+        }
+
+        if (count($config) == 1 && array_key_exists('dsn', $config)) {
```

Comment:
dsn option could be used with other options in config. It should work like [here](https://github.com/php-enqueue/enqueue-dev/blob/master/pkg/amqp-tools/ConnectionConfig.php#L137)
  
The config is treat as default options, and options from dsn could overwrite them